index.bs

```diff
@@ -1141,6 +1243,218 @@ steps:
+these steps:
+
+1. If [=this=].[=[[state]]=] is "closed", throw a {{InvalidStateError}}.
+1. If  |newSize| is less than 0, throw a {{InvalidStateError}}.
+1. Let |fileContents| be a copy of [=this=].[=[[file]]=]'s [=file entry/binary data=].
+1. Let |p| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |oldSize| be the [=byte sequence/length=] of [=this=].[=[[file]]=]'s [=file entry/binary data=].
+  1. If |newSize| is larger than |oldSize|:
+     1. If |newSize| - |oldSize| exceeds the available [=storage quota=], [=/reject=] |p|
+        with a {{QuotaExceededError}} and abort.
+     1. Set [=this=].[=[[file]]=]'s to a [=byte sequence=] formed by concating
+        |fileContents| with a [=byte sequence=]
+        containing |newSize|-|oldSize| `0x00` bytes.
+  1. Else if |newSize| is smaller than |oldSize|:
+     1. Set [=this=].[=[[file]]=]'s to a [=byte sequence=] containing the first |newSize| bytes
+        in |fileContents|.
+  1. [=/Resolve=] |p|.
+1. Return |p|
+
+</div>
+
+### The {{FileSystemSyncAccessHandle/getSize()}} method ### {#api-filesystemsyncaccesshandle-getsize}
+
+<div class="note domintro">
+  : |handle| . {{FileSystemSyncAccessHandle/getSize()}}
+  :: Returns the size of the file associated with |handle| in bytes.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemSyncAccessHandle>getSize()</dfn> method, when invoked, must run
+these steps:
+
+1. If [=this=].[=[[state]]=] is "closed", throw a {{InvalidStateError}}.
+1. Let |p| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |size| be the [=byte sequence/length=] of [=this=].[=[[file]]=]'s [=file entry/binary data=].
+  1. [=/Resolve=] |p| with |size|.
+1. Return |p|
+
+
+</div>
+
+### The {{FileSystemSyncAccessHandle/flush()}} method ### {#api-filesystemsyncaccesshandle-flush}
+
+<div class="note domintro">
+  : |handle| . {{FileSystemSyncAccessHandle/flush()}}
+  :: Ensures that the contents of the file associated with |handle| contain all the modifications done through {{FileSystemSyncAccessHandle/write()}}.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemSyncAccessHandle>flush()</dfn> method, when invoked, must run
+these steps:
+
+// TODO(fivedots): Fill in, after figuring out language to describe flushing at the OS level.
+
+</div>
+
+### The {{FileSystemSyncAccessHandle/close()}} method ### {#api-filesystemsyncaccesshandle-close}
+
+<div class="note domintro">
+  : |handle| . {{FileSystemSyncAccessHandle/close()}}
+  :: Flushes the access handle and then closes it. Closing an access handle disables any further operations on it and
+     [=file entry/lock/release|releases the lock=] on the [=FileSystemHandle/entry=] associated with |handle|.
+</div>
+
+//TODO(fivedots): Figure out language to describe flushing the file at the OS level before closing the handle.
+<div algorithm>
+The <dfn method for=FileSystemSyncAccessHandle>close()</dfn> method, when invoked, must run
+these steps:
```

Comment:
Method steps

Reply:
This looks like a preexisting pattern in the spec. I'll leave it as is and file a cleanup bug on the whatwg/fs repo.